plugins/devtools-backend/src/service/router.ts

```diff
@@ -35,30 +35,66 @@ import { createPermissionIntegrationRouter } from '@backstage/plugin-permission-
 import {
   DiscoveryService,
   HttpAuthService,
+  LifecycleService,
+  LoggerService,
   PermissionsService,
 } from '@backstage/backend-plugin-api';
+import { SignalService } from '@backstage/plugin-signals-node';
 
 /** @public */
 export interface RouterOptions {
   devToolsBackendApi?: DevToolsBackendApi;
-  logger: Logger;
+  logger: LoggerService;
   config: Config;
   permissions: PermissionsService;
   discovery: DiscoveryService;
   httpAuth?: HttpAuthService;
+  lifecycle?: LifecycleService;
+  signalService?: SignalService;
 }
 
 /** @public */
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
-  const { logger, config, permissions } = options;
+  const { logger, config, permissions, lifecycle, signalService } = options;
 
   const { httpAuth } = createLegacyAuthAdapters(options);
 
   const devToolsBackendApi =
     options.devToolsBackendApi || new DevToolsBackendApi(logger, config);
 
+  if (lifecycle && signalService) {
```

Comment:
I do not have a strong opinion on requiring lifecycle here, but it should probably be documented somewhere that both have to be defined as it might not be intuitive.

Reply:
With the new backend system these are both defined automatically. For old backend the lifecycle is not available so might just drop support for that

Reply:
good point - thanks for pointing this out! this should be good!:)